ee/desktop/notify/notify_darwin.m

```diff
@@ -18,13 +18,20 @@ BOOL doSendNotification(UNUserNotificationCenter *center, NSString *title, NSStr
         content:content trigger:nil];
 
     __block BOOL success = YES;
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
 
-    [center addNotificationRequest:request withCompletionHandler:^(NSError * _Nullable error) {
-        if (error != nil) {
-            NSLog(@"Could not send notification: %@", error);
-            success = NO;
-        }
-    }];
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        [center addNotificationRequest:request withCompletionHandler:^(NSError * _Nullable error) {
+            if (error != nil) {
+                NSLog(@"Could not send notification: %@", error);
+                success = NO;
+            }
+            dispatch_semaphore_signal(semaphore);
+        }];
+    });
+
+    // Wait for completion handler to complete so that we get a correct value for `success`
+    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
```

Comment:
Potentially waiting forever scares me, should we use a reasonable timeout? Something like if the notification can't be sent within 30 seconds, we consider it failed and call `removePendingNotificationRequestsWithIdentifiers`?

Reply:
Yup that's absolutely reasonable, I've updated to 30 seconds per suggestion! We shouldn't need to do anything beyond that, `success` will be `NO` still and the error will get reported back up and handled 🙂

Reply:
The error will be handled, but if the timeout gets hit then we would still have a pending notification in the `UNUserNotificationCenter` queue and it could be delivered after the fact. I think that could allow a weird sequence where each attempt we try to schedule the notification, it doesn't get sent in time, we think it failed, but it did actually show.

Reply:
Oh, sorry, I misunderstood your initial suggestion! I think I've got it updated appropriately now.